adagrams/game.py

```diff
@@ -1,11 +1,75 @@
+import copy
+import random
+from operator import indexOf
+
+LETTERS = {'A': 9, 'B' : 2, 'C' : 2, 'D' : 4, 'E' : 12, 'F' : 2, 'G' : 3, 'H' : 2, 'I' : 9, 'J' : 1, 'K' : 1, 'L' : 4,
+            'M': 2, 'N' : 6, 'O' : 8, 'P' : 2, 'Q' : 1, 'R' : 6, 'S' : 4, 'T' : 6, 'U' : 4, 'V' : 2, 'W' : 2, 'X' : 1, 'Y' : 2, 'Z' : 1}
+
+SCORE_CHART = {'A' : 1, 'B' : 3, 'C' : 3, 'D' : 2, 'E' : 1, 'F' : 4, 'G' : 2, 'H' : 4, 'I' : 1, 'J' : 8, 'K' : 5, 'L' : 1, 'M' : 3, 'N' : 1,
+                'O' : 1, 'P' : 3, 'Q' : 10, 'R' : 1, 'S' : 1, 'T' : 1, 'U' : 1, 'V' : 4, 'W' : 4, 'X' : 8, 'Y' : 4, 'Z' : 10}
+
 def draw_letters():
-    pass
+    copied_letters = copy.deepcopy(LETTERS)
+    hand = []
+    while len(hand) < 10:
+        # Get a random letter among the keys in LETTERS dictionary
+        rand_letter = random.choice(list(LETTERS.keys()))
+        # Add letter in hand only if we have enough number of letters in copied letter pool
+        if copied_letters[rand_letter] > 0:
+            hand.append(rand_letter)
+            # Substract one after adding a letter to the hand
+            copied_letters[rand_letter] -= 1
+    return hand
```

Comment:
I believe that while your function won't give too many of a given letter it doesn't follow the proper distribution of letters.

Since you pick all keys with equal probability you're as likely to draw a Z or a X as your first letter as you are to draw an A or an E.

This is a really subtle bug though, and don't feel bad you missed it!